main.py

```
@@ -22,11 +22,123 @@
+class _UpdateChecker(QThread):
+    """Queries the GitHub releases API on a background thread."""
+    update_available = pyqtSignal(str, str)  # (tag, html_url)
+    up_to_date = pyqtSignal()
+
+    def run(self):
+        try:
+            url = f"https://api.github.com/repos/{_GITHUB_REPO}/releases/latest"
+            req = urllib.request.Request(
+                url,
+                headers={"Accept": "application/vnd.github+json", "User-Agent": "JobDocs"},
+            )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)